src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,79 @@
+import { resetCurrentTodo, selectCurrentTodo } from '../../features';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoUser, setTodoUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  const activeTodo = useAppSelector(selectCurrentTodo);
+
+  useEffect(() => {
+    if (activeTodo) {
+      setIsLoading(true);
+      getUser(activeTodo.userId)
+        .then(setTodoUser)
+        .finally(() => setIsLoading(false));
+    } else {
+      setIsLoading(false);
+      setTodoUser(null);
+    }
+  }, [activeTodo]);
+
+  if (!activeTodo) {
+    return null;
+  }
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${activeTodo.id}`}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(resetCurrentTodo())}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {activeTodo.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              {activeTodo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {todoUser && (
+                <a href={`mailto:${todoUser.email}`}>{todoUser.name}</a>
+              )}
```

Comment:
Consider handling the case where `todoUser` might be null more explicitly to avoid potential rendering issues. Ensure that the email and name are available before rendering the link.